core/src/main/java/org/infinispan/stats/impl/AbstractClusterStats.java

```diff
@@ -0,0 +1,162 @@
+package org.infinispan.stats.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.infinispan.commons.CacheException;
+import org.infinispan.jmx.JmxStatisticsExposer;
+import org.infinispan.jmx.annotations.DataType;
+import org.infinispan.jmx.annotations.DisplayType;
+import org.infinispan.jmx.annotations.ManagedAttribute;
+import org.infinispan.jmx.annotations.ManagedOperation;
+import org.infinispan.jmx.annotations.Parameter;
+import org.infinispan.jmx.annotations.Units;
+import org.infinispan.util.TimeService;
+import org.infinispan.util.logging.Log;
+
+/**
+ * @author Ryan Emerson
+ * @since 9.0
+ */
+public abstract class AbstractClusterStats implements JmxStatisticsExposer {
+
+   public static final long DEFAULT_STALE_STATS_THRESHOLD = 3000;
+
+   protected TimeService timeService;
+   private boolean statisticsEnabled = false;
+   private long staleStatsTreshold = DEFAULT_STALE_STATS_THRESHOLD;
+   private long statsUpdateTimestamp = 0;
+
+   private final Log log;
+   private final AtomicLong resetNanoseconds = new AtomicLong(0);
+   final HashMap<String, Number> statsMap = new HashMap<>();
```

Comment:
This has to be package private or protected as it is accessed in `ClusterCacheStatsImpl::updateTimeSinceStart` and `ClusterCacheStatsImpl::updateStats`